includes/sanitizers/class-amp-bento-sanitizer.php

```diff
@@ -229,53 +230,144 @@ private function get_bento_component_name_from_url( $url ) {
 	 * @param Element $amp_element AMP element (converted from Bento).
 	 */
 	private function adapt_layout_styles( Element $amp_element ) {
+		$supported_layouts = [];
+		foreach ( AMP_Allowed_Tags_Generated::get_allowed_tag( $amp_element->tagName ) as $rule_spec ) {
+			if ( isset( $rule_spec['tag_spec']['amp_layout']['supported_layouts'] ) ) {
+				$supported_layouts = array_merge( $supported_layouts, $rule_spec['tag_spec']['amp_layout']['supported_layouts'] );
+			}
+		}
+		$supported_layouts = array_unique( $supported_layouts );
+
+		// If no layouts are supported, then there is nothing to do.
+		if ( count( $supported_layouts ) === 0 ) {
+			return;
+		}
+
+		// If nodisplay is the only supported layout or nodisplay is supported the
+		// element is hidden, then give it the nodisplay layout.
+		if (
+			[ Layout::TO_SPEC[ Layout::NODISPLAY ] ] === $supported_layouts
+			||
+			(
+				$amp_element->hasAttribute( Attribute::HIDDEN )
+				&&
+				in_array( Layout::TO_SPEC[ Layout::NODISPLAY ], $supported_layouts )
```

Comment:
Do we want it to be a strict check?

```suggestion
				in_array( Layout::TO_SPEC[ Layout::NODISPLAY ], $supported_layouts, true )
```

Reply:
Yes, good catch.